conans/tools.py

```diff
@@ -221,6 +222,18 @@ def latest_vs_version_installed(*args, **kwargs):
     return tools_win.latest_vs_version_installed(output=_global_output, *args, **kwargs)
 
 
+def shell_quote(*args):
```

Comment:
Unless explicitly required, these things doesn't go to "tools" as they would get exposed in conanfile.py recipes. If needed in code, put that in the "generators" package. In any case, this seems a bit complex, not sure what problems it is addressing.

Reply:
Shell paths must be quoted (escaped). `shlex.sh_quote()` just not working for Windows platform. What solution do you propose?